addons/mail/models/mail_compose_message.py

```diff
@@ -643,17 +666,101 @@ def _gc_lost_attachments(self):
             ('write_date', '<', limit_date)]
         ).unlink()
 
+    @api.autovacuum
+    def _gc_not_scheduled(self):
+        """ Garbage collect composers that are not scheduled (drafts that were not scheduled nor
+        sent, and sent composers) """
+        limit_date = fields.Datetime.subtract(fields.Datetime.now(), days=1)
+        self.search([('state', '!=', 'scheduled'), ('write_date', '<', limit_date)]).unlink()
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        composers = super().create(vals_list)
+        # link attachments to composer (they were created with res_id=0)
+        for composer in composers:
+            if attachments := composer.attachment_ids:
+                attachments.filtered(
+                    lambda a: a.res_model == self._name and not a.res_id
+                ).write({'res_id': composer.id})
+        return composers
+
+    def write(self, vals):
+        if any(composer.state == 'sent' for composer in self):
+            raise UserError(_("Sent composers cannot be modified"))
+
+        res = super().write(vals)
+
+        # (re)schedule cron trigger if needed
+        trigger_datetimes = set()
+        if scheduled_date := vals.get('scheduled_date'):
+            if any(composer.state == 'scheduled' for composer in self):
+                scheduled_datetime = fields.Datetime.to_datetime(scheduled_date)
+                if scheduled_datetime < fields.Datetime.now():
+                    raise UserError(_("Cannot schedule a message in the past."))
```

Comment:
Can't we consider putting a datetime in the past = scheduling for now = trigger for now ? Not costly and allows to skip an error.